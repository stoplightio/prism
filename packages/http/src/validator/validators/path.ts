import { HttpParamStyles, IHttpPathParam } from '@stoplight/types';
import { IHttpNameValue } from '../../types';
import { IHttpParamDeserializerRegistry } from '../deserializers/types';
import { HttpParamsValidator } from './params';

export class HttpPathValidator extends HttpParamsValidator<IHttpNameValue> {
  constructor(
    registry: IHttpParamDeserializerRegistry<IHttpNameValue>,
    prefix: string,
    style: HttpParamStyles = HttpParamStyles.Simple
  ) {
    super(registry, prefix, style);
  }
  public validate(target: IHttpNameValue, specs: IHttpPathParam[]) {
    return super.validate(target, specs);
  }
}
