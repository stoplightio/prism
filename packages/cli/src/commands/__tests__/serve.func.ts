import { resolve } from 'path';
import Serve from '../serve';

describe('serve command', () => {
  it('throws error when supplied with non-existing path', async () => {
    return expect(Serve.run(['-s', 'a-non-existing-path'])).rejects.toThrowErrorMatchingSnapshot();
  });

  it('throws error when supplied with a directory', async () => {
    return expect(Serve.run(['-s', resolve('examples')])).rejects.toThrowErrorMatchingSnapshot();
  });
});
