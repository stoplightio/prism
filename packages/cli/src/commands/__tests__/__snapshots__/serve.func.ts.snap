// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serve command throws error when supplied with a directory 1`] = `"Supplied spec path points to directory. Only files are supported."`;

exports[`serve command throws error when supplied with non-existing path 1`] = `"Non-existing path to spec supplied: a-non-existing-path"`;
